src/main/java/org/opensearch/jobscheduler/sweeper/JobSweeper.java

```diff
@@ -229,6 +230,15 @@ public void postDelete(ShardId shardId, Engine.Delete delete, Engine.DeleteResul
                     exception -> log.debug("Failed to delete lock", exception)
                 )
             );
+        } else if (this.scheduler.getDeScheduledJobIds(shardId.getIndexName()).contains(delete.id())) {
```

Comment:
Oh I see, there is a bug when deleting a disabled job that it may not also delete corresponding lock?

Maybe we should combine this with the block above since the code within the block is duplicated?